urlcb91e63e47_alexGalushka_E97_Assignment3_tgz-pJ8-cscie97_asn3_test_TestDriverJ8/src/cscie97/asn3/squaredesk/renter/SchedulingService.java

```diff
@@ -0,0 +1,269 @@
+package cscie97.asn3.squaredesk.renter;
+
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import cscie97.asn2.squaredesk.provider.OfficeSpace;
+
+public class SchedulingService
+{
+
+	private static SchedulingService _obj;
+	private Map<String, List<Booking>> bookingMap;
+	private SimpleDateFormat sdf;
+	private String date;
+	private String[] dateArray;
+	private Calendar calendar;
+	private int FIX_NUM_DAYS = 95;
+	
+	/**
+	 * unparameterized constructor: created a brand new bookingMap with FIX_NUM_DAYS keys
+	 */
+	private SchedulingService()
+	{
+		bookingMap = new HashMap<String,List<Booking>>();
+		sdf = new SimpleDateFormat("yyyy M d");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.